ghost/core/test/e2e-webhooks/site.test.js

```diff
@@ -0,0 +1,57 @@
+const {agentProvider, mockManager, fixtureManager, matchers} = require('../utils/e2e-framework');
+const {anyGhostAgent, anyContentVersion, anyNumber} = matchers;
+
+describe('site.* events', function () {
+    let adminAPIAgent;
+    let webhookMockReceiver;
+
+    before(async function () {
+        adminAPIAgent = await agentProvider.getAdminAPIAgent();
+        await fixtureManager.init('integrations');
+        await adminAPIAgent.loginAsOwner();
+    });
+
+    beforeEach(function () {
+        webhookMockReceiver = mockManager.mockWebhookRequests();
+    });
+
+    afterEach(function () {
+        mockManager.restore();
+    });
+
+    it('site.changed event is triggered', async function () {
+        const webhookURL = 'https://test-webhook-receiver.com/site-changed';
+        await webhookMockReceiver.mock(webhookURL);
+        await fixtureManager.insertWebhook({
+            event: 'site.changed',
+            url: webhookURL
+        });
+
+        const res = await adminAPIAgent
+            .post('posts/')
+            .body({
+                posts: [{
+                    title: 'webhookz',
+                    status: 'published',
+                    mobiledoc: fixtureManager.get('posts', 1).mobiledoc
+                }]
+            })
+            .expectStatus(201);
+
+        const id = res.body.posts[0].id;
+
+        await adminAPIAgent
```

Comment:
Deleting a post is not necessary here. Publishing a post with a title "webhookz" doest the job too. Less code is better code 😊 

The 'site.changed' event is tiggered when Ghost [sends a cache invalidation header](https://github.com/TryGhost/Ghost/blob/ab598dd7ea3d1cc82f80bec5d0d04780a7edfe2e/ghost/core/core/server/web/parent/middleware/emit-events.js#L8-L10), which is triggered when [the post is published](https://github.com/TryGhost/Ghost/blob/ab598dd7ea3d1cc82f80bec5d0d04780a7edfe2e/ghost/core/core/server/api/endpoints/posts.js#L123-L127).

Reply:
Ah good catch, I missed this. Should be fixed now.